src/mqt/problemsolver/partialcompiler/qaoa.py

```diff
@@ -119,8 +138,41 @@ def remove_unnecessary_gates(self, qc: QuantumCircuit, optimize_swaps: bool = Tr
 
         qc._data = [v for i, v in enumerate(qc._data) if i not in indices]
 
+        if self.satellite_use_case:
+            return self.apply_factors_to_qc(qc)
+
         return qc
 
+    def apply_factors_to_qc(self, qc: QuantumCircuit) -> QuantumCircuit:
+        """Applies factors to each qubit representing the location image value and the dependencies to other image locations."""
+        # create QUBO formulation based on interactions of between qubits
+        qubo = QuadraticProgramToQubo().convert(from_docplex_mp(self.create_model_from_pair_list()))
+        # extract the factors: one for each qubit/image location, one factor for all ZZ interactions, one factor for all mixer layers
+        coeffs = np.array(qubo.to_ising()[0].primitive.coeffs, dtype=float)
+        coeffs_qubits = coeffs[: self.num_qubits]
+        coeffs_interactions = coeffs[self.num_qubits + 1]
+        coeff_mixer = qubo.to_ising()[1]
```

Comment:
It could make sense to extract the `to_ising()` call so that it is not repeated.

Reply:
Good point, I improved that and merge the PR.